packages/tools/package/src/models/package/package.ts

```diff
@@ -93,6 +93,40 @@ class Package {
       .then(() => this);
   }
 
+  /**
+   * Apply the modifications of this Package instance to the package definition
+   * files.
+   *
+   * @returns - Promse that resolves to this Package instance.
+   */
+  public syncDependency(deps: Record<string, string>, devDeps: boolean = false): Promise<this> {
```

Comment:
`syncDependencies` may not be needed. when we run `yarn npm publish` on a sub-package using **yarn workspaces** it looks for the `version` key:value in the `package.json` for all dependent packages automatically and writes them for us. We should only need to put the correct and current `version` key:value into each sub-packages `package.json`, and not touch the dependencies list within each package.

Reply:
I thought so too. But I was confused if semantic-release did this for us all this while or the `yarn npm publish`

Thanks for the clarification @InteractiveTimmy

Reply:
@InteractiveTimmy @mkesavan13 I've removed syncDependency based on the discussion